cdap-app-templates/cdap-etl/cdap-etl-realtime/src/test/java/co/cask/cdap/template/etl/realtime/RealtimeCubeSinkTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.template.etl.realtime;
+
+import co.cask.cdap.api.dataset.lib.cube.AggregationFunction;
+import co.cask.cdap.api.dataset.lib.cube.Cube;
+import co.cask.cdap.api.dataset.lib.cube.CubeQuery;
+import co.cask.cdap.api.dataset.lib.cube.TimeSeries;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data2.dataset2.lib.cube.CubeDatasetDefinition;
+import co.cask.cdap.proto.AdapterConfig;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.template.etl.api.PipelineConfigurable;
+import co.cask.cdap.template.etl.api.realtime.RealtimeSource;
+import co.cask.cdap.template.etl.common.ETLStage;
+import co.cask.cdap.template.etl.common.Properties;
+import co.cask.cdap.template.etl.realtime.config.ETLRealtimeConfig;
+import co.cask.cdap.template.etl.realtime.sink.RealtimeCubeSink;
+import co.cask.cdap.template.etl.realtime.sink.RealtimeTableSink;
+import co.cask.cdap.template.etl.realtime.sink.StreamSink;
+import co.cask.cdap.template.etl.realtime.source.JmsSource;
+import co.cask.cdap.template.etl.realtime.source.KafkaSource;
+import co.cask.cdap.template.etl.realtime.source.TestSource;
+import co.cask.cdap.template.etl.realtime.source.TwitterSource;
+import co.cask.cdap.template.etl.transform.ProjectionTransform;
+import co.cask.cdap.template.etl.transform.ScriptFilterTransform;
+import co.cask.cdap.template.etl.transform.StructuredRecordToGenericRecordTransform;
+import co.cask.cdap.test.AdapterManager;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.cdap.test.TestBase;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.gson.Gson;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ *
+ */
+public class RealtimeCubeSinkTest extends TestBase {
```

Comment:
no. Instead, as I mentioned, ETLWorkerTest should be broken down into classes!

It is separate and this one will differ from ETLWorkerTest. It has similar one method due to the bug we discussed